### PR TITLE
Adding CQL statement tracing for Connector30

### PR DESCRIPTION
**Pull request by nivancevic**

### CODE REVIEW

File: cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java [hunk 1]
```diff
@@ -22,27 +23,32 @@
 
     private static Cluster cluster;
     private static Session session;
-    private static CountDownLatch lock = new CountDownLatch(1);
-    private static boolean queryIntercepted;
+    private static CountDownLatch lock = new CountDownLatch(2);
+    private static int queryInterceptedCnt;
```
Comment (nbozic-zz) [inline cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java]: I would rather spend two more letters and make this clear `queryInterceptedCount`

File: cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java [hunk 2]
```diff
@@ -22,27 +23,32 @@
...
+                        query.statement().equalsIgnoreCase("SELECT uid FROM test_keyspace.test_table")) {
+                    queryInterceptedCnt++;
                     lock.countDown();
                 }
             }
         }, configuration);
+        
         EmbeddedCassandraServerHelper.startEmbeddedCassandra();
         connector.waitForSetupCompleted();
+        Thread.sleep(4000);
```
Comment (nbozic-zz) [inline cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java:61]: Do we have any better idea to get rid of bunch of Thread.sleep stuff from tests? Not related to this PR

File: cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java [hunk 3]
```diff
@@ -52,10 +58,14 @@ public void query_is_intercepted_when_connector_is_active() throws InterruptedEx
...
+        String cql = "SELECT uid FROM test_keyspace.test_table";
+        PreparedStatement prepared = session.prepare(cql);
+        session.execute(cql);
+        session.execute(prepared.bind());
+        Thread.sleep(2000);
         cluster.close();
-        lock.await(2000, TimeUnit.MILLISECONDS);
-        Assert.assertTrue(queryIntercepted);
+        lock.await(60000, TimeUnit.MILLISECONDS);
+        Assert.assertEquals(2, queryInterceptedCnt);
```
Comment (nbozic-zz) [inline cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java]: I had hard time following why 2 and realized that you are testing both prepared and plain string in same method.

Does it make sense to divide into two tests or to rename to `both_prepared_and_string_statement_intercepted_when_connector_is_active` or something like that.

File: cassandra-diagnostics-connector30/src/integration-test/resources/cu-cassandra.yaml [hunk 4]
```diff
@@ -412,20 +412,20 @@ compaction_throughput_mb_per_sec: 16
...
-cas_contention_timeout_in_ms: 1000
+cas_contention_timeout_in_ms: 10000
 # How long the coordinator should wait for truncates to complete
 # (This can be much longer, because unless auto_snapshot is disabled
 # we need to flush first so we can snapshot before removing the data.)
-truncate_request_timeout_in_ms: 60000
+truncate_request_timeout_in_ms: 600000
 # The default timeout for other, miscellaneous operations
-request_timeout_in_ms: 10000
+request_timeout_in_ms: 100000
```
Comment (nbozic-zz) [inline cassandra-diagnostics-connector30/src/integration-test/resources/cu-cassandra.yaml]: Was this tweak while you were testing or we need to increase this config that much?

**Comment by nbozic-zz** [comment]
Also missing README, we need to add enableTracing configuration and to explain that this is for debugging purposes, that people need to be aware of performance impact etc.

File: cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java (hunk 2 above)
Comment (nivancevic) [reply cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java:61]: Not yet. Too much time spent determining when Cassandra is really really ready to accept statements from client.

Comment (nbozic-zz) [reply cassandra-diagnostics-connector30/src/integration-test/java/io/smartcat/cassandra/diagnostics/connector/ITConnector.java:61]: Can we create an issue at least. I see this growing throughout the code.

**Review by nbozic-zz — APPROVED** [review]

**Pull request merged**